src/ui/TextField.h

```diff
@@ -30,6 +30,7 @@ class TextField : public QLineEdit
         void setLabelFontSize(qreal size);
         void setShowLabel(bool value);
         void setUnderlineColor(const QColor &color);
+        void setRegexp(const QString &regexp);
```

Comment:
Why not store a QRegularExpression?

Reply:
Ok, will try to add it

Reply:
Please also make this take a QRegularExpression :3

Reply:
Oh, okay)